tests/js/client/server_parameters/test-telemetrics-true-auth.js

```diff
@@ -467,6 +472,44 @@ function telemetricsShellReconnectGraphTestsuite() {
         }
       }
     },
+
+    testTelemetricsShellExecuteScriptLeave: function () {
+      // this is for when the user logs into the shell to run a script and then leave the shell immediately. The shell
+      // should not hang because telemetrics is still in progress, either by fetching telemetrics info from servers or
+      // by sending it to the endpoint so the shell process must be finished just right after the script execution ends.
+      let file = fs.getTempFile() + "-telemetrics";
+      fs.write(file, `(function() { const x = 0;})();`);
+      let options = internal.options();
+      let endpoint = arango.getEndpoint().replace(/\+vpp/, '').replace(/^http:/, 'tcp:').replace(/^https:/, 'ssl:').replace(/^vst:/, 'tcp:').replace(/^h2:/, 'tcp:');
+      const args = {
+        'javascript.startup-directory': options['javascript.startup-directory'],
+        'server.endpoint': endpoint,
+        'server.username': arango.connectedUser(),
+        'server.password': '',
+        'javascript.execute': file,
+        'client.failure-points': 'startTelemetricsForTest',
+      };
+      const argv = toArgv(args);
+
+      for (let o in options['javascript.module-directory']) {
+        argv.push('--javascript.module-directory');
+        argv.push(options['javascript.module-directory'][o]);
+      }
+
+      let result = internal.executeExternal(arangosh, argv, false /*usePipes*/);
+      assertTrue(result.hasOwnProperty('pid'));
+      let numSecs = 0.5;
+      let status = {};
+      while (true) {
+        status = internal.statusExternal(result.pid);
+        if (status.status === "TERMINATED" || numSecs >= 16) {
+          break;
+        }
+        internal.sleep(numSecs);
+        numSecs *= 2;
+      }
+      assertEqual(status.status, "TERMINATED", "couldn't leave shell immediately after executing script");
```

Comment:
Readers of this code may wonder what this test actually checks.
It is unclear from this code that here we want the start sub-shell to close quickly **despite** any telemetrics requests or connection setups being in flight. There is also no mention here that the connection setup could take up to 30 seconds and we want to be able to interrupt it ASAP if the shell is otherwise ready to close.